enterprise/cypher/physical-planning/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/SlotAllocation.scala

```diff
@@ -469,6 +473,9 @@ object SlotAllocation {
         lhs.newReference(collectionName.name, nullable, CTList(CTAny))
         lhs
 
+      case _: ForeachApply =>
+        lhs
```

Comment:
It's not clear to me why slot allocation for ForeachApply is not done here. It's done on line 516 instead.

Reply:
The slot for the inner variable has to be included in the argument for the RHS. The way slot allocation currently works the whole RHS slot configuration have already been allocated when we come here.